src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,218 @@
+        }
+        String time = elements[1];
+        String textDescription = elements[0];
+        Task event = new Event(textDescription, time);
+        listOfItems.addTask(event);
+        return ui.printAdditionConfirmation(event, listOfItems);
+    }
+
+    /**
+     * Adds a deadline item to the list.
+     *
+     * @param input Gui input from the user.
+     * @return String confirmation of task being successfully added.
+     * @throws DukeException when input format is wrong.
+     */
+    private String addAsDeadline(String input) throws DukeException {
+        String[] keywords = input.split(" ", 2);
+        if (keywords.length == 1) {
+            throw new DukeException("you need to describe your deadline to me in format: deadline (description)!");
+        }
+        String[] elements = keywords[1].split(" /by ", 2);
+        if (elements.length == 1) {
+            throw new DukeException("you need to tell me the time in format: (task) /by (time)!");
+        }
+        String time = elements[1];
+        String textDescription = elements[0];
+        Task deadline = new Deadline(textDescription, time);
+        listOfItems.addTask(deadline);
+        return ui.printAdditionConfirmation(deadline, listOfItems);
+    }
+
+    /**
+     * Deletes an item from the list by passing the index of the task to be deleted.
+     *
+     * @param input Command line input from the user to be parsed.
+     * @return String confirmation of task being successfully deleted.
+     * @throws DukeException Throws an exception if index is not on the list or with wrong input.
+     */
+    private String deleteItem(String input) throws DukeException {
+        try {
+            Integer idx = Integer.parseInt(parser.getDescription(input));
+            Task item = listOfItems.getTask(idx);
+            listOfItems.removeTask(idx);
+            return ui.printDeletionConfirmation(item, listOfItems);
+        } catch (NumberFormatException e) {
+            throw new DukeException("you need to choose a number from the list in the form: delete (list index)!");
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("you need to pick an item from within the list!");
+        }
+    }
+
+    /**
+     * Displays a list of items on the command line that contain the keyword entered by the user.
+     *
+     * @param input Command line input from the user.
+     * @return String list of items on the task list.
+     * @throws DukeException Throws an exception if no keyword was entered or if no tasks match the keyword given.
+     */
+    private String findItems(String input) throws DukeException {
+        try {
+            String keyword = parser.getDescription(input);
+            Tasklist validItems = listOfItems.findAllBy(keyword);
+            return ui.listToPrintableString(validItems);
+        }catch (ArrayIndexOutOfBoundsException e){
```

Comment:
You may add a space between the `}` and the `catch` statement as well as `(` and `{`.
```suggestion
        } catch (ArrayIndexOutOfBoundsException e) {
```